### PR TITLE
Profile page, and main page filtering

### PR DESCRIPTION
**Pull request by Colby-M**
This adds the profile page petitions and solutions to look through

Also adds in the main page filtering by scope. This is not finished yet but should be close.

I will pull in main after @Kyle-Mie merges his other branches in first

### CODE REVIEW

File: APolitcalApp/src/views/ViewProfilePage.vue [hunk 1]
```diff
@@ -13,8 +36,23 @@
...
     <template #ContentSlot>
-      <h1>profile</h1>
-      <AuthAccount @sign-out="authStore.logout()" v-if="authStore.session" :session="authStore.session" />
+      <div class="flex flex-col gap-5">
+        <AuthAccount @sign-out="authStore.logout()" v-if="authStore.session" :session="authStore.session" />
+        <div class="flex w-full justify-between">
+          <div>
+            <p>Your Petitions</p>
+            <div class="flex flex-col gap-2">
+              <CardPetitionTitle v-for="petition in petitions" :key="petition.id" :petition-id="petition.id" :petition-title="petition.title" />
```
Comment (Kyle-Mie) [inline APolitcalApp/src/views/ViewProfilePage.vue:54]: When clicking on a petition it redirects to url.com/review/petitionId, but the review page is only setup to handle url.com/review?id=petitionId

File: APolitcalApp/src/views/ViewProfilePage.vue [hunk 2]
```diff
@@ -13,8 +36,23 @@
...
+            </div>
+          </div>
+          <div>
+            <p>Your Solutions</p>
+            <div class="flex flex-col gap-2">
+              <CardSolutionsOtherSuggestions v-for="post in solutions" :linkedPetition="post.id" :suggestionText="post.description" :uservotes="post.uservotes.length ?? 0" />
```
Comment (Kyle-Mie) [inline APolitcalApp/src/views/ViewProfilePage.vue]: I am getting the console error TypeError: Cannot read properties of null (reading 'length'); and none of the solutions render. Not sure why the ?? isn't working.

**Review by Kyle-Mie — CHANGES_REQUESTED** [review]

File: APolitcalApp/src/views/ViewProfilePage.vue (hunk 2 above)
Comment (Colby-M) [reply APolitcalApp/src/views/ViewProfilePage.vue]: I don't actually have any comments yet so I didn't notice anything wrong but I will look

File: APolitcalApp/src/views/ViewProfilePage.vue (hunk 1 above)
Comment (Colby-M) [reply APolitcalApp/src/views/ViewProfilePage.vue:54]: I reused this component but I will look

**Comment by Colby-M** [comment]
Going to need to create new component or something for showing our own solutions in the profile page. There is no link right now to the original petition.

Also what does approving a petition do? shouldn't you have to review the solutions inside to approve those and not the petition itself

**Comment by Kyle-Mie** [comment]
> Going to need to create new component or something for showing our own solutions in the profile page. There is no link right now to the original petition.
> 
> Also what does approving a petition do? shouldn't you have to review the solutions inside to approve those and not the petition itself

For now I just linked approving the petition to approving the solution they proposed in the description. I figure once we put in a timer or something to select the top solution I can change the UI to make it more clear to the user.

**Review by Kyle-Mie — APPROVED** [review]

**Pull request merged**